rafx/examples/renderer_triangle/features/jobs/extract.rs

```diff
@@ -0,0 +1,66 @@
+use crate::features::internal::{
+    DemoFramePacket, DemoPerFrameData, DemoRenderFeatureTypes, DemoRenderObjectStaticData,
+};
+use crate::TimeState;
+use rafx::assets::{AssetManagerRenderResource, MaterialAsset};
+use rafx::distill::loader::handle::Handle;
+use rafx::framework::render_features::ExtractJob;
+use rafx::render_feature_extract_job_predule::*;
+use rafx_base::resource_map::ReadBorrow;
+use rafx_base::resource_ref_map::ResourceRefBorrow;
+use std::sync::Arc;
+
+pub struct DemoExtractJob<'extract> {
+    time_state: ResourceRefBorrow<'extract, TimeState>,
+    asset_manager: ReadBorrow<'extract, AssetManagerRenderResource>,
+    triangle_material: Handle<MaterialAsset>,
+}
+
+impl<'extract> DemoExtractJob<'extract> {
+    pub fn new(
+        extract_context: &RenderJobExtractContext<'extract>,
+        frame_packet: Box<DemoFramePacket>,
+        triangle_material: Handle<MaterialAsset>,
+    ) -> Arc<dyn RenderFeatureExtractJob<'extract> + 'extract> {
+        Arc::new(ExtractJob::new(
+            Self {
+                time_state: extract_context.extract_resources.fetch::<TimeState>(),
+                asset_manager: extract_context
+                    .render_resources
+                    .fetch::<AssetManagerRenderResource>(),
+                triangle_material,
+            },
+            frame_packet,
+        ))
+    }
+}
+
+impl<'extract> ExtractJobEntryPoints<'extract> for DemoExtractJob<'extract> {
+    fn begin_per_frame_extract(
+        &self,
+        _render_objects: &RenderObjectsMap<DemoRenderObjectStaticData>,
+        context: &ExtractPerFrameContext<'extract, '_, Self>,
+    ) {
+        context
```

Comment:
I think having an example that sets up the jobs is a great idea. Maybe an extremely basic particle system where we extract some positions, produce some sprites around each position, and then draw them. This way we use a little more of the job system and show how per-entity stuff works. This can be a separate thing we do in the future. Happy for this to go in as it is now.

Reply:
Agree with making a more detailed example later. This example is also just _incredibly_ helpful when making changes to the jobs. Only having to update a single feature is a life saver on iteration times.

Reply:
Tracking in https://github.com/aclysma/rafx/issues/137